wurst/math/Matrices.wurst

```diff
@@ -0,0 +1,310 @@
+        this.rc20*scalar, this.rc21*scalar, this.rc22*scalar)
+
+/** Matrix multiplication is not commutative that means matrix*vect is not the same as vec*matrix. */
+public function real.op_mult(mat3 m) returns mat3
+    return m*this
+
+/** Matrix multiplication is not commutative that means matrix*vect is not the same as vec*matrix. */
+public function mat3.op_mult(vec3 v) returns vec3
+    return vec3(
+        this.rc00*v.x + this.rc01*v.y + this.rc02*v.z,
+        this.rc10*v.x + this.rc11*v.y + this.rc12*v.z,
+        this.rc20*v.x + this.rc21*v.y + this.rc22*v.z)
+
+public function vec3.op_mult(mat3 m) returns vec3
+    return vec3(
+        this.x*m.rc00 + this.y*m.rc10 + this.z*m.rc20,
+        this.x*m.rc01 + this.y*m.rc11 + this.z*m.rc21,
+        this.x*m.rc02 + this.y*m.rc12 + this.z*m.rc22)
+
+public function mat3.op_mult(mat3 m) returns mat3
+    // row by row
+    return mat3(
+        this.rc00*m.rc00 + this.rc01*m.rc10 + this.rc02*m.rc20,
+        this.rc00*m.rc01 + this.rc01*m.rc11 + this.rc02*m.rc21,
+        this.rc00*m.rc02 + this.rc01*m.rc12 + this.rc02*m.rc22,
+
+        this.rc10*m.rc00 + this.rc11*m.rc10 + this.rc12*m.rc20,
+        this.rc10*m.rc01 + this.rc11*m.rc11 + this.rc12*m.rc21,
+        this.rc10*m.rc02 + this.rc11*m.rc12 + this.rc12*m.rc22,
+
+        this.rc20*m.rc00 + this.rc21*m.rc10 + this.rc22*m.rc20,
+        this.rc20*m.rc01 + this.rc21*m.rc11 + this.rc22*m.rc21,
+        this.rc20*m.rc02 + this.rc21*m.rc12 + this.rc22*m.rc22)
+    
+/** Returns matrix' column by id as a vector or zero-vector if doesn't exist. */
+public function mat3.col(int id) returns vec3
+    switch id
+        case 0
+            return vec3(this.rc00, this.rc10, this.rc20)
+        case 1
+            return vec3(this.rc01, this.rc11, this.rc21)
+        case 2
+            return vec3(this.rc02, this.rc12, this.rc22)
+        default
+            return ZERO3
+
+/** Returns matrix' row by id as a vector or zero-vector if doesn't exist. */
+public function mat3.row(int id) returns vec3
+    switch id
+        case 0
+            return vec3(this.rc00, this.rc01, this.rc02)
+        case 1
+            return vec3(this.rc10, this.rc11, this.rc12)
+        case 2
+            return vec3(this.rc20, this.rc21, this.rc22)
+        default
+            return ZERO3
+        
+public function mat3.determinant() returns real
+    return this.rc00*(this.rc11*this.rc22 - this.rc12*this.rc21)
+         - this.rc01*(this.rc10*this.rc22 - this.rc12*this.rc20)
+         + this.rc02*(this.rc10*this.rc21 - this.rc11*this.rc20)
+
+/** Flips the matrix over its main diagonale.
+Result of transposing for rotation matrix is its inverse matrix. */
+public function mat3.transpose() returns mat3
+    return mat3(
+        this.rc00, this.rc10, this.rc20,
+        this.rc01, this.rc11, this.rc21,
+        this.rc02, this.rc12, this.rc22)
+
+/** Returns a zero-matrix in case there's no iverse matrix.*/

```

Comment:
Doc comment should describe more than just it's corner case.
Also, consider adding a `.hasInverse()` function instead of returning a zero-matrix, or returning a result tuple like
`inverseresutl(boolean success, mat3 result)`.

Trying to check if this call succeeded will otherwise entail 9 comparison if you want to compare it against the zero-matrix.

Reply:
For such kind of test we have to calc determinant and once more in ``.inverse`` if it's not 0. More operations that's undesirable.  I think tuple-result is the best solution thanks for hint.